### PR TITLE
Allow incubating Panama Vector in simdvec, and add vectorized ipByteBin

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Add support for vectorized ipByteBin.

The structure of the implementation and loading framework mirror that of Lucene, but is simplified by avoiding reflective loading since ES has support for a MRJar section for 21. 

For now, we just disable warnings-as-errors in this small sourceset, since `-Xlint:-incubating` is only support since JDK 22. The number of source files is small here. Will investigate how to assert that just the single incubating warning is emitted by javac, at a later point.

### CODE REVIEW

File: libs/simdvec/build.gradle [hunk 1]
```diff
@@ -23,6 +23,19 @@ dependencies {
   }
 }
 
+tasks.named("compileMain21Java").configure {
+  options.compilerArgs << '--add-modules=jdk.incubator.vector'
+  // we remove Werror, since incubating suppression (-Xlint:-incubating)
+  // is only support since JDK 22
+  options.compilerArgs -= '-Werror'
```
Comment (ChrisHegarty) [inline libs/simdvec/build.gradle:30]: @breskeby as discussed off-line, this small change disables warnings-as-error for this small sourceset. I think that the change is inline with what we discussed?

File: libs/simdvec/src/main/java/org/elasticsearch/simdvec/ESVectorUtil.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package org.elasticsearch.simdvec;
+
+import org.elasticsearch.simdvec.internal.vectorization.ESVectorUtilSupport;
+import org.elasticsearch.simdvec.internal.vectorization.ESVectorizationProvider;
+
+import static org.elasticsearch.simdvec.internal.vectorization.ESVectorUtilSupport.B_QUERY;
+
+public class ESVectorUtil {
```
[the remark this answers is not in the document]
Comment (ChrisHegarty) [reply libs/simdvec/src/main/java/org/elasticsearch/simdvec/ESVectorUtil.java:17]: Hmm... this is a good point. Both classes exist for somewhat similar reasons - temporay workaround the lack of support in Lucene. Lemme see how best to unify these in a follow up.

File: libs/simdvec/src/main21/java/org/elasticsearch/simdvec/internal/vectorization/ESVectorizationProvider.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    "Java vector incubator API enabled; uses preferredBitSize=%d",
+                    PanamaESVectorUtilSupport.VECTOR_BITSIZE
+                )
+            );
+            return impl;
+        } else {
+            logger.warn(
+                "You are running with unsupported Java "
+                    + runtimeVersion
+                    + ". To make full use of the Vector API, please update Apache Lucene."
```
Comment (benwtrent) [inline libs/simdvec/src/main21/java/org/elasticsearch/simdvec/internal/vectorization/ESVectorizationProvider.java:86]: nit, Apache Lucene isn't the issue here.

**Review by benwtrent — APPROVED** [review]
From my perspective, this is LGTM. Minor logging issue and I am fine with unifying our "vector util" stuff into a single class later.

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**